File: RunnabelandHandler/app/src/main/java/com/resatkarakaya/runnabelandhandler/MainActivity.java
package com.resatkarakaya.runnabelandhandler;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import static java.lang.Thread.*;

public class MainActivity extends AppCompatActivity {

    TextView textView;
    int number;
    Runnable runnable;
    Handler handler;
    Button button;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        button = findViewById(R.id.button);
        number=0;
    }

    public void start(View view){

        handler = new Handler();

        runnable = new Runnable() {
            @Override
            public void run() {
                textView.setText("Time: " + number);
                number++;
                textView.setText("Time :" + number);
                handler.postDelayed(runnable,1000);
            }
        };
        handler.post(runnable);
        button.setEnabled(false); // butonu bir kere tıkladıktan sonra kapattı
    }

    public void stop(View view){
        button.setEnabled(true);
        handler.removeCallbacks(runnable);
        number = 0;
        textView.setText("Time: " + number);
    }
}

File: Kotlin/KotlinRunnable/app/src/main/java/com/resatkarakaya/kotlinrunnable/MainActivity.kt
package com.resatkarakaya.kotlinrunnable

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var number = 0
    var runnable : Runnable = Runnable {  }
    var handler : Handler = Handler()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun start(view : View){
        //while (number < 100){
       //number++
       // textView.text = "Time: $number"
        //Thread.sleep(1000)
       // } Doğru değil Runnable kullanılması gerekir

        number = 0

        runnable = object : Runnable {
            override fun run() {
                number++
                textView.text="Time: $number"

                handler.postDelayed(this,1000)
            }
        }
    }

    fun stop(view : View){
        handler.removeCallbacks(runnable)
        number = 0
        textView.text = "Time: 0"
    }
}